Source/WebCore/html/HTMLSummaryElement.cpp

```diff
@@ -101,7 +103,23 @@ static bool isClickableControl(EventTarget* target)
     if (!is<Element>(target))
         return false;
     auto& element = downcast<Element>(*target);
-    return is<HTMLFormControlElement>(element) || is<HTMLFormControlElement>(element.shadowHost());
+    if (is<HTMLElement>(element)) {
+        for (auto& ancestorOrSelf : lineageOfType<HTMLElement>(element)) {
+            if (is<HTMLSummaryElement>(ancestorOrSelf))
+                return false;
+            if (ancestorOrSelf.isInteractiveContent())
+                return true;
+        }
+        return false;
+    }
+    if (is<SVGElement>(element)) {
+        for (auto& ancestorOrSelf : lineageOfType<SVGElement>(element)) {
+            if (is<SVGAElement>(ancestorOrSelf))
+                return true;
+        }
+        return false;
+    }
```

Comment:
What should happen if you have SVG in HTML or vice-versa? Would be good to have a spec reference to clarify those things :)

Reply:
> What should happen if you have SVG in HTML or vice-versa?

The only SVG element that has activation behavior is the SVG `a` element. The HTML spec doesn’t explicitly state that, because the HTML spec doesn’t state anything explicitly about the behavior of particular SVG elements.

But the only way you can have an SVG `a` element inside a `summary` is if there’s a `svg` element descendant of the `summary`, so that sets the SVG namespace, and then an SVG `a` descendant of that.

As far as HTML in SVG, I think the only relevant way that could be done in this context is by having an `svg` descendant of the `summary` and then a `foreignObject` descendant of that `svg` element, and then some HTML interactive content (with activation behavior) inside that `foreignObject`.

That `foreignObject` case seemed to me like too much of a pathological case to try to deal with — it’s hard to imagine any legitimate real-world need to ever mark up some `summary` content using that.

But if we want to handle the `foreignObject` case, I guess we could do that `lineageOfType` check using `Element` as the type, and then inside there, do an `is<HTMLElement>` type check of each ancestorOrSelf, and then a cast before calling `isInteractiveContent()`. It’d just be uglier and catching the `foreignObject` corner case would be the only reason for the additional complexity and runtime cost. 

But aside from that, what would help to simplify the existing code in the patch is if the spec explicitly defined the SVG `a` element as “interactive content”. I already chatted with Anne about that and I’ll be raising a spec PR tomorrow to make that spec change. (That change actually would also have normative implementation impact on `label` handling.) But the current patch in this branch would get us passing all the relevant WPT tests — without regressing anything — so I was thinking (hoping) we could possibly land this first, and deal with implementing the SVG-`a`-element-as-interactive-content change after this has landed first, and after the spec change lands.

> Would be good to have a spec reference to clarify those things :)

The spec isn’t explicit about the requirements for how interactive content works when inside `summary`. It seems to only be defined implicitly, by the places in the spec that state what the “activation behavior” is for any elements that have “activation behavior”. And what’s implicit about it is: the activation behavior of elements that are descendants of `summary` take precedence over the activation behavior of `summary` itself. I guess ideally the spec should be updated to state that explicitly.

Reply:
OK, I'm fine addressing this in a follow up then, since this is interoperable. Regarding the spec, I was hoping the summary activation behavior would reflect this check too: https://html.spec.whatwg.org/multipage/interactive-elements.html#the-summary-element

Reply:
> Regarding the spec, I was hoping the summary activation behavior would reflect this check too: [html.spec.whatwg.org/multipage/interactive-elements.html#the-summary-element](https://html.spec.whatwg.org/multipage/interactive-elements.html#the-summary-element)

I’m happy to add that, but shall we do handle it with a separate follow-up PR? For one thing, it seems we may not have any tests for it — so I reckon I’d also need to write tests for it.

Reply:
Yeah that's fine!

Reply:
> What should happen if you have SVG in HTML or vice-versa? Would be good to have a spec reference to clarify those things :)

The tests I added now include testing the HTML-in-SVG case — inside `foreignObject`. And in the existing tests we already had tests for the SVG-in-HTML case. So it seems we got coverage for all the SVG-HTML combination cases now (and the patch in this PR branch is handling them all as expected).